### DISCUSSION TITLE
A query related to baseurl in hugo doks theme

### DISCUSSION THREAD

**Discussion by beartester34** [thuliteio/doks#1277] [Q&A] [answered]
Hello everyone. I am new to HUGO and I hope you guys will bear with me on my query.

I installed VSCODE and Git on my mac.

I make my website on my mac, then i sync it with Github and then I deploy the latest commit on Render. 

I made a website with hugo doks theme and then hosted it on Render

Then I used a custom domain https://tetert.site/

Everything is good till here

I then go to https://pagespeed.web.dev and I get an error (in fact two errors)

1st error

`<span jsslot="">
Line # | Content | Error
-- | -- | --
3 | Sitemap: /sitemap.xml | Invalid sitemap URL

</span>
`

and 2nd error

`Document does not have a valid rel=canonicalIs not an absolute URL (/)
Canonical links suggest which URL to show in search results. [Learn more about canonical links](https://developer.chrome.com/docs/lighthouse/seo/canonical/?utm_source=lighthouse&utm_medium=lr).`

So, I saw the source code of my website

it has
`<base href=[/](view-source:https://tetert.site/)><link rel=canonical href=[/](view-source:https://tetert.site/)><link rel=alternate type=application/rss+xml href=[/index.xml](view-source:https://tetert.site/index.xml) title="My Docs">`

I tried changing the baseurl in hugo.toml file to https://tetert.site/
but still, nothing happens

the `<base href=[/](view-source:https://tetert.site/)><link rel=canonical href=[/](view-source:https://tetert.site/)><link rel=alternate type=application/rss+xml href=[/index.xml](view-source:https://tetert.site/index.xml) title="My Docs">`
`

is still the same

whereas in the website getdoks.org

the base is defined as `<base href=[https://getdoks.org/](view-source:https://getdoks.org/)><link rel=canonical href=[https://getdoks.org/](view-source:https://getdoks.org/)>`

what am I doing wrong ?

**Comment by h-enk** [comment, marked as answer]
Did you update `config/production/hugo.toml` for your production URL?

It should be like this:

```toml
# Overrides for production environment
baseurl = "https://tetert.site/"
```

**Reply by h-enk** [reply]
For local development, keep the original setting in `config/_default/hugo.toml`:

```toml
baseurl = "http://localhost/"
```

**Reply by beartester34** [reply]
> ```toml
> http://localhost/
> ```

Wow

It works
Thanks a lot
